pkg/deploy/controller/deployer/factory.go

```diff
@@ -217,6 +217,7 @@ func (c *DeploymentController) getByKey(key string) (*v1.ReplicationController,
 	}
 	rc, err := c.rcLister.ReplicationControllers(namespace).Get(name)
 	if errors.IsNotFound(err) {
+		// TODO tnozicka: this is not normal and should be refactored
```

Comment:
What is the value in having this function now?  I expected to see something like this: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L553-L561